datarepo-clienttests/src/main/java/scripts/testscripts/RetrieveSnapshot.java

```diff
@@ -34,14 +41,54 @@ public RetrieveSnapshot() {
 
   private SnapshotSummaryModel snapshotSummaryModel;
   private List<BlobId> scratchFiles = new ArrayList<>();
+  private boolean deleteProfile;
 
   public void setup(List<TestUserSpecification> testUsers) throws Exception {
-    // create the profile and dataset
-    super.setup(testUsers);
+    // Specify the name of a profile to use for the test with the TEST_RUNNER_BILLING_PROFILE_NAME
+    // env variable.
+    // If it's not specified or the profile isn't found, a new one with a randomly generated name
+    // will be created
+    String profileNameRaw = System.getenv("TEST_RUNNER_BILLING_PROFILE_NAME");
+    Optional<String> profileName;
+    if (StringUtils.isBlank(profileNameRaw)) {
+      profileName = Optional.empty();
+      deleteProfile = true;
+      logger.info("No profile name specified.  Will create a new billing profile");
+    } else {
+      profileName = Optional.ofNullable(profileNameRaw);
+      deleteProfile = false;
+      logger.info("Will attempt to use a billing profile named {}", profileName.get());
+    }
+
+    // pick the a user that is a Data Repo steward to be the dataset creator
+    datasetCreator = SAMUtils.findTestUserThatIsDataRepoSteward(testUsers, server);
 
     // get the ApiClient for the snapshot creator, same as the dataset creator
     ApiClient datasetCreatorClient = DataRepoUtils.getClientForTestUser(datasetCreator, server);
     RepositoryApi repositoryApi = new RepositoryApi(datasetCreatorClient);
+    ResourcesApi resourcesApi = new ResourcesApi(datasetCreatorClient);
+    BillingProfileModel billingProfileModel =
+        profileName
+            .map(
+                p -> {
+                  try {
+                    List<BillingProfileModel> profiles =
+                        resourcesApi.enumerateProfiles(0, 100).getItems();
+
+                    BillingProfileModel foundProfile =
+                        IterableUtils.find(
+                            profiles, m -> StringUtils.equalsIgnoreCase(p, m.getProfileName()));
```

Comment:
I am curious why you're using profileName instead of ID. It seems like we've gotten into trouble before by selecting on names instead of UUIDs.

Reply:
I was thinking it might be a little more meaningful. I actually did a first pass with uuid but part of me was thinking that would allow us to have consistent naming on the different environments.  I don't feel super strongly about it though. I'm happy to change it to an ID

Reply:
Eh, I might leave the name for now (really easy to change to ID).  Just seems easier to work with

Reply:
Gotcha, makes sense!